homeassistant/components/home_connect/select.py

```diff
@@ -459,6 +453,28 @@ def update_native_value(self) -> None:
             data.value
         )
 
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        setting = self.appliance.settings.get(cast(SettingKey, self.bsh_key))
+        if (
+            not setting
+            or not setting.constraints
+            or not setting.constraints.allowed_values
+        ):
+            with contextlib.suppress(HomeConnectError):
+                setting = await self.coordinator.client.get_setting(
```

Comment:
Side note: I think we'll have to refactor this in the future so that all client requests stem from coordinator methods so that we can keep track of the requests and not cross the rate limit.

Reply:
That sounds right, I would like to address that before 2025.3

Reply:
It's only 3 days until beta cut on Wednesday, so that may be hard.